Base VIPER Interfaces/BaseWireframe.swift

```diff
@@ -1,46 +1,58 @@
 import UIKit
 
-enum Transition {
-    case root
-    case push
-    case present(fromViewController: UIViewController)
-}
-
 protocol WireframeInterface: class {
     func popFromNavigationController(animated: Bool)
     func dismiss(animated: Bool)
 }
 
 class BaseWireframe {
 
-    unowned var navigationController: UINavigationController
+    private unowned var _viewController: UIViewController
+    private var vcDealloc: UIViewController?
 
-    init(navigationController: UINavigationController) {
-        self.navigationController = navigationController
+    init(viewController: UIViewController) {
+        _viewController = viewController
+        vcDealloc = viewController
```

Comment:
Better naming like: `_storedViewController` and add some comment why it is needed (to retain view controller reference upon first access)

Reply:
Agree on this one, vcDealloc is a really shitty name :D